### PR TITLE
Falkon: add version 3.1.0

### PR DESCRIPTION
**Pull request by digrouz**

### CODE REVIEW

File: bucket/falkon.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+    "homepage": "https://www.falkon.org",
+    "version": "3.1.0",
+    "description": "Falkon is a KDE web browser using QtWebEngine rendering engine, previously known as QupZilla.",
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
    "description": "KDE web browser using QtWebEngine rendering engine, previously known as QupZilla.",
```

File: bucket/falkon.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "license": {
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
    "license": "GPL-3.0-or-later",
```

File: bucket/falkon.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        }
+    },
+    "bin": "falkon.exe",
+    "shortcuts": [
+        [
+            "falkon.exe",
+            "Falkon"
+        ]
+    ],
+    "checkver": {
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
    "checkver": "Current version: ([\\d.]+)"
```

File: bucket/falkon.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        "url": "https://www.falkon.org/",
+        "checkver": "Current version: ([\\d.]+)"
+    },
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://download.kde.org/stable/falkon/$version/Falkon.Installer.$version.x64.exe#/dl.7z"
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
                "url": "https://download.kde.org/stable/falkon/$majorVersion.$minorVersion/Falkon.Installer.$version.x64.exe#/dl.7z"
```

File: bucket/falkon.json [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+            },
+            "32bit": {
+                "url": "https://download.kde.org/stable/falkon/$version/Falkon.Installer.$version.exe#/dl.7z"
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
                "url": "https://download.kde.org/stable/falkon/$majorVersion.$minorVersion/Falkon.Installer.$version.exe#/dl.7z"
```

File: bucket/falkon.json [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+            }
+        }
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
        "hash": {
            "url": "$url.sha256"
        }
```

image

**Comment by Ash258** [comment]
There will most likely be lots of https -> http redirection as majority of mirrors are http

**Comment by digrouz** [comment]
do you want to change the downloads to http ?

**Comment by Ash258** [comment]
I do not know if it will help

**Comment by digrouz** [comment]
I guess it's best to leave the redirects do their work to avoid managing a link to a mirror

**Comment by digrouz** [comment]
Anything I can do to finalize that pull request ?

**Comment by Ash258** [comment]
What about fixing manifest?

File: bucket/falkon.json [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{
+    "homepage": "https://www.falkon.org",
+    "version": "3.1.0",
+    "description": "KDE web browser using QtWebEngine rendering engine, previously known as QupZilla.",
+    "license": "GPL-3.0-or-later",
+        "identifier": "GPL-3.0-or-later",
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
```

File: bucket/falkon.json [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+        "url": "https://phabricator.kde.org/source/falkon/browse/master/COPYING"
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
```

File: bucket/falkon.json [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        }
+    },
+    "bin": "falkon.exe",
+    "shortcuts": [
+        [
+            "falkon.exe",
+            "Falkon"
+        ]
+    ],
+    "checkver":
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
```

File: bucket/falkon.json [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        "url": "https://www.falkon.org/",
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
```

File: bucket/falkon.json [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        "checkver": "Current version: ([\\d.]+)"
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
    "checkver": "Current version: ([\\d.]+)",
```

File: bucket/falkon.json [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    },
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
```

File: bucket/falkon.json [hunk 13]
```diff
@@ -0,0 +1,39 @@
+{
+    "homepage": "https://www.falkon.org",
+    "version": "3.1.0",
+    "description": "KDE web browser using QtWebEngine rendering engine, previously known as QupZilla.",
+    "license": "GPL-3.0-or-later",
+    },
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
```

File: bucket/falkon.json [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+            "64bit": {
+                "url": "https://download.kde.org/stable/falkon/$majorVersion.$minorVersion/Falkon.Installer.$version.x64.exe#/dl.7z"
+            },
+            "32bit": {
+                "url": "https://download.kde.org/stable/falkon/$majorVersion.$minorVersion/Falkon.Installer.$version.exe#/dl.7z"
+            },
+            "hash": {
+              "url": "$url.sha256"
+            }
+        }
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
        },
        "hash": {
            "url": "$url.sha256"
        }
```

File: bucket/falkon.json [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+    "checkver": "Current version: ([\\d.]+)",
+    "autoupdate": {
+        "architecture": {
+            "64bit": {
+                "url": "https://download.kde.org/stable/falkon/$majorVersion.$minorVersion/Falkon.Installer.$version.x64.exe#/dl.7z"
+            },
+            "32bit": {
+                "url": "https://download.kde.org/stable/falkon/$majorVersion.$minorVersion/Falkon.Installer.$version.exe#/dl.7z"
+            },
+            "hash": {
```
Comment (Ash258) [inline bucket/falkon.json]: Remove this property. You moved it into wrong scope

Comment (digrouz) [reply bucket/falkon.json]: done

**Comment by digrouz** [comment]
seems fixed now, thanks for the help

**Comment by digrouz** [comment]
now we just wait someone to merge ?

File: bucket/falkon.json [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+    "architecture": {
+        "64bit": {
+            "url": "https://download.kde.org/stable/falkon/3.1/Falkon.Installer.3.1.0.x64.exe#/dl.7z",
+            "hash": "5ab4c139dee19c21114f7acbe38c40babb392bfb875cb977db11fd390b7aa7ee"
+        },
+        "32bit": {
+            "url": "https://download.kde.org/stable/falkon/3.1/Falkon.Installer.3.1.0.exe#/dl.7z",
+            "hash": "806dc30fd68edee4587e2f480054000f82d5d56df5047ee299547cb2247a3a88"
+        }
+    },
```
Comment (Ash258) [inline bucket/falkon.json]:
```suggestion
    },
    "pre_install": "Remove-Item \"$dir\\`$PLUGINSDIR\", \"$dir\\uninstall.exe\" -Recurse -Force",
```

**Comment by dodorz** [comment]
The 32bit exe file's hash is "3b82d1a4661192b3d98c155bffc6a29605748be9ee0b974a8632f383abc88ebd".

**Comment by digrouz** [comment]
> The 32bit exe file's hash is "3b82d1a4661192b3d98c155bffc6a29605748be9ee0b974a8632f383abc88ebd".

nope, if you look at https://download.kde.org/stable/falkon/3.1/Falkon.Installer.3.1.0.exe.sha256 you get `806dc30fd68edee4587e2f480054000f82d5d56df5047ee299547cb2247a3a88  Falkon.Installer.3.1.0.exe`

**Comment by digrouz** [comment]
@Ash258 Do you know if something is preventing the merge ?

**Pull request merged**